plugins/opa-permissions-wrapper/src/permission-handler/permissionHandler.ts

```diff
@@ -19,19 +21,39 @@ export class PermissionsHandler {
       )}`,
     );
 
-    const makePolicyDecision = createOpaPermissionEvaluator(this.opaClient);
+    if (isResourcePermission(request.permission, 'catalog-entity')) {
+      this.logger.info('Catalog Permission Request') // Debugging for now
+      const makeCatalogPolicyDecision = catalogPolicyEvaluator(
+        this.opaClient
+      );
+      const policyDescision = await makeCatalogPolicyDecision(request, user);
 
-    const policyDecision = await makePolicyDecision(request, user);
+      return policyDescision;
+    }
 
-    this.logger.info(
-      `Policy decision: ${JSON.stringify(
-        policyDecision,
-      )} for user: ${JSON.stringify(user)} and request: ${JSON.stringify(
-        request,
-      )}`,
-    );
+    if (isResourcePermission(request.permission, 'scaffolder-action')) {
+      this.logger.info('Scaffolder Action Permission Request') // Debugging for now
+      const makeScaffolderActionPolicyDecision = scaffolderActionPolicyEvaluator(
+        this.opaClient
+      );
+      const policyDescision = await makeScaffolderActionPolicyDecision(request, user);
+
+      return policyDescision;
+    }
+
+    if (isResourcePermission(request.permission, 'scaffolder-template')) {
+      this.logger.info('Scaffolder Template Permission Request') // Debugging for now
+      const makeScaffolderTemplatePolicyDecision = scaffolderTemplatePolicyEvaluator(
+        this.opaClient
+      );
+      const policyDescision = await makeScaffolderTemplatePolicyDecision(request, user);
+
+      return policyDescision;
+    }
+
+    this.logger.error('Unknown permission type');
+    throw new Error('Unknown permission type');
```

Comment:
I _think_ this is a good approach right now, but may need to think about this.

If you check the `*Evaluator.ts` files they all have different methods you need to call depending on the resource type.